app/src/main/java/com/asfoundation/wallet/topup/TopUpFragment.kt

```diff
@@ -428,18 +432,17 @@ class TopUpFragment : DaggerFragment(), TopUpFragmentView {
   }
 
   private fun buildBonusString(bonus: BigDecimal, bonusCurrency: String) {
-    var scaledBonus = bonus.stripTrailingZeros()
-        .setScale(2, BigDecimal.ROUND_FLOOR)
+    var scaledBonus = bonus
     var currency = bonusCurrency
     if (scaledBonus < BigDecimal(0.01)) {
       currency = "~$currency"
     }
     scaledBonus = scaledBonus.max(BigDecimal("0.01"))
```

Comment:
We're doing unnecessary work here.
it can be replaced with :
`    val scaledBonus = bonus.max(BigDecimal("0.01"))
    val currency = "~$bonusCurrency".takeIf { bonus < BigDecimal("0.01") } ?: bonusCurrency`